testing/src/main/java/net/consensys/linea/reporting/TestOutcomeWriterTool.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package net.consensys.linea.reporting;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ConcurrentSkipListSet;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import net.consensys.linea.zktracer.json.JsonConverter;
+
+@Slf4j
+public class TestOutcomeWriterTool {
+
+  public static void addFailed() {
+    failedCounter.incrementAndGet();
+  }
+
+  public static JsonConverter jsonConverter = JsonConverter.builder().build();
+  private static ObjectMapper objectMapper = new ObjectMapper();
+
+  private static volatile AtomicInteger failedCounter = new AtomicInteger(0);
+  private static volatile AtomicInteger successCounter = new AtomicInteger(0);
+  private static volatile AtomicInteger disabledCounter = new AtomicInteger(0);
+  private static volatile AtomicInteger abortedCounter = new AtomicInteger(0);
+  private static volatile ConcurrentMap<
+          String, ConcurrentMap<String, ConcurrentSkipListSet<String>>>
+      modulesToConstraintsToTests = new ConcurrentHashMap<>(20);
+
+  public static void addFailure(String type, String cause, String test) {
+    failedCounter.incrementAndGet();
+    modulesToConstraintsToTests
+        .computeIfAbsent(type, t -> new ConcurrentHashMap<>())
+        .computeIfAbsent(cause, t -> new ConcurrentSkipListSet<>())
+        .add(test);
+  }
+
+  public static void addSuccess() {
+    successCounter.incrementAndGet();
+  }
+
+  public static void addAborted() {
+    abortedCounter.incrementAndGet();
+  }
+
+  public static void addSkipped() {
+    disabledCounter.incrementAndGet();
```

Comment:
It's strange that we talk about `disabled` and `skipped` rather than having one word for both. Presumably `disabled` makes the most sense if it in any way involves the `@Disabled` annotation (which I have no idea if it does or not)

Reply:
it comes from the interface i implement. So I guess it's standard.